crates/flox-rust-sdk/src/models/publish.rs

```diff
@@ -754,4 +897,168 @@ mod tests {
 
         assert!(matches!(result, Err(ConvertFlakeRefError::NoRemote(_))));
     }
+
+    /// Check if we successfully clone a repo
+    #[tokio::test]
+    async fn upstream_repo_from_url() {
+        let _ = env_logger::try_init();
+
+        let (_flox, temp_dir_handle) = flox_instance();
+        let repo_dir = temp_dir_handle.path().join("repo");
+
+        // create a repo
+        fs::create_dir(&repo_dir).unwrap();
+        let repo = Git::init(&repo_dir, false).await.unwrap();
+
+        UpstreamRepo::clone_repo(
+            repo.workdir().unwrap().to_string_lossy(),
+            temp_dir_handle.path(),
+        )
+        .await
+        .expect("Should clone repo");
+    }
+
+    // disabled because nix build does not have git user/email config,
+    // TODO fix tests to work with local repos
+    #[cfg(feature = "impure-unit-tests")]
+    /// Check if we successfully clone a repo
+    #[tokio::test]
+    async fn create_catalog_branch() {
+        let _ = env_logger::try_init();
+        let (_flox, temp_dir_handle) = flox_instance();
+        let repo_dir = temp_dir_handle.path().join("repo");
+
+        // create a repo
+        fs::create_dir(&repo_dir).unwrap();
+        let repo = Git::init(&repo_dir, false).await.unwrap();
+        let mut repo = UpstreamRepo::clone_repo(
+            repo.workdir().unwrap().to_string_lossy(),
+            temp_dir_handle.path(),
+        )
+        .await
+        .expect("Should clone repo");
+
+        assert!(repo.0.list_branches().await.unwrap().is_empty());
+
+        let catalog = repo
+            .get_or_create_catalog(&"aarch64-darwin".to_string())
```

Comment:
Would it be better to use `flox.system` instead of hardcoding? Probably doesn't really matter